src/vm/nativeformatreader.h

```diff
@@ -482,6 +486,60 @@ namespace NativeFormat
 
         bool IsNull() { return _pReader == NULL; }
 
+        class AllEntriesEnumerator
```

Comment:
I'm not sure I see a need for this class.  Does the `Enumerator` class on line 547 skip entries in the hash table?  It seems like this does effectively the same thing as that class, but returns a `NativeParser` rather than using an out variable.

Reply:
`Enumerator` conceptually only enumerates entries in one bucket, where `AllEntriesEnumerator` enumerates everything in the hashtable. It would be possible to modify `Enumerator` to be able to enumerate all entries, but with this code I am not sure there is much value. I like having the intent clear, either you are enumerating one bucket or all buckets.

Reply:
Ah, I see it know; `Enumerator` is only intended to be used as part of a call to `Lookup`.  I agree with your assessment.  In fact, I almost wish the names were flipped, but that isn't necessary/worth the effort.

Reply:
The `Enumerator` and `AllEntriesEnumerator` class also exist the way they're written here in the dotnet/corert repo. We had this code since the .Net Native release.